### PR TITLE
[security] Add security assessment commands

### PR DESCRIPTION
**Pull request by chlahav**
**Description**  

Adding assessment management cmdlets to az security module
- assessment - list, show, create, delete
- assessment-metadata - list, show, create, delete
- sub-assessment - list, show

**Testing Guide**  
Assessments and sub assessments are generated by Microsoft.Security RP if there are any relevant insights on your subscription.
AssessmentMetadata shows all the available assessment types and by default returns the built in assessment types.

Users can create their own managed assessment by creating a metadata first:
   `az security assessment-metadata create -n "8f211ea8-777b-45c4-8631-776d535afd62" --display-name "test assessment" --severity "High" --description "hello world"`
and then create assessments on specific resources:
   `az security assessment create -n 8f211ea8-777b-45c4-8631-776d535afd62 --status-code Healthy`
When a metadata is deleted, all the assessments from this type is deleted as well

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
security

File: src/azure-cli/azure/cli/command_modules/security/custom.py [hunk 1]
```diff
@@ -324,10 +331,129 @@ def update_atp_setting(client, resource_group_name, storage_account_name, is_ena
...
+
+    return client.list(scope='/subscriptions/' + client.config.subscription_id)
+
+
+def get_security_assessment(client, resource_name, assessed_resource_id=None):
+
+    if assessed_resource_id is None:
+        assessed_resource_id = '/subscriptions/' + client.config.subscription_id
+
+    print(assessed_resource_id)
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/security/custom.py:357]: Is it intentional or forget to delete?

Comment (chlahav) [reply src/azure-cli/azure/cli/command_modules/security/custom.py]: sorry, it was a mistake
removed

**Pull request merged**